### ISSUE TITLE
TerraformTaskV1: Terraform plan executed twice

### ISSUE THREAD

**Issue by Jeppeks1** [microsoft/azure-pipelines-terraform#32]
Hello,

I recently stumbled upon this Terraform extension to Azure Pipelines and thought I would try implementing it in my deployments. I was surprised to find that the terraform plan command is executed twice instead of just once, which increases the time it takes to run the pipeline. 

Would it be possible to modify the task so the plan command is executed once? For example if the -out parameter is present in the first plan command, the existing terraform plan can be used to produce the JSON output and therefore eliminating the need for a second plan step?

**Comment by mericstam** [comment]
should work now.  Still facing issues? reopen the issue if you do.

**Comment by xenalite** [comment]
@mericstam Hello. Please re-open this issue. I have managed to reproduce using both TerraformTaskV1@0 (version resolved to 0.203.0) and TerraformTaskV2@2 (version resolved to 2.203.0).

Repro pipeline:
```
name: $(Rev:rr)
trigger:
- main
variables:
  system.debug: true
  tf_dir: $(Build.SourcesDirectory)/terraform
  azure_connection: <redacted>

steps:
- task: TerraformTaskV2@2
  displayName: init
  inputs:
    provider: azurerm
    command: init
    workingDirectory: $(tf_dir)
    backendServiceArm: $(azure_connection)
    backendAzureRmResourceGroupName: <redacted>
    backendAzureRmStorageAccountName: <redacted>
    backendAzureRmContainerName: terraform-state-files
    backendAzureRmKey: omtest.tfstate

- task: TerraformTaskV2@2
  displayName: plan
  inputs:
    provider: azurerm
    command: plan
    workingDirectory: $(tf_dir)
    environmentServiceNameAzureRM: $(azure_connection)
```

Task log output (V2):
[7.txt](https://github.com/microsoft/azure-pipelines-terraform/files/8588131/7.txt)

You can see in the debug log that terraform plan command runs twice with different parameters:
1:
```
2022-04-29T04:19:31.0918453Z ##[debug]exec tool: /usr/local/bin/terraform
2022-04-29T04:19:31.0919238Z ##[debug]arguments:
2022-04-29T04:19:31.0920008Z ##[debug]   plan
2022-04-29T04:19:31.0921103Z ##[debug]   -detailed-exitcode
2022-04-29T04:19:31.0921798Z [command]/usr/local/bin/terraform plan -detailed-exitcode
2022-04-29T04:19:32.9006741Z 
2022-04-29T04:19:32.9035932Z [0m[1m[32mNo changes.[0m[1m Your infrastructure matches the configuration.[0m
2022-04-29T04:19:32.9036513Z 
2022-04-29T04:19:32.9037379Z [0mTerraform has compared your real infrastructure against your configuration
2022-04-29T04:19:32.9038182Z and found no differences, so no changes are needed.
```

2:
```
2022-04-29T04:19:33.0580368Z ##[debug]exec tool: /usr/local/bin/terraform
2022-04-29T04:19:33.0581372Z ##[debug]arguments:
2022-04-29T04:19:33.0582132Z ##[debug]   plan
2022-04-29T04:19:33.0583386Z ##[debug]   -out=/home/vsts/work/1/s/plan-binary-d799e9be-e8f4-4f0a-bb9f-f607e169b205.tfplan
```

Initially I thought it's just a mistake in the logs so I decided to test how many times plan gets called by using side-effects.
I added a http data resource to terraform file to call an Azure Function and log the number of requests during pipeline by Application Insights.
```
data "http" "example" {
  url = "https://<redacted>.azurewebsites.net/api/HttpTrigger1?code=<redacted>==&name=test2"
}
```

![image](https://user-images.githubusercontent.com/4509839/165885327-914bc9b5-b341-46b4-a968-3874663326f0.png)

**Comment by mericstam** [comment]
Hi, V3 release is out in version 0.1.17 on the Marketplace.
